adserver/staff/forms.py

```diff
@@ -0,0 +1,171 @@
+
+    def __init__(self, *args, **kwargs):
+        """Add the form helper and customize the look of the form."""
+        super().__init__(*args, **kwargs)
+        self.helper = FormHelper()
+
+        self.helper.layout = Layout(
+            Fieldset(
+                _("Advertiser information"),
+                Field("advertiser_name", placeholder=_("Company name")),
+                css_class="my-3",
+            ),
+            Fieldset(
+                _("Managing user"),
+                Field("user_name"),
+                Field("user_email", placeholder="advertiser@company.com"),
+                css_class="my-3",
+            ),
+            Submit("submit", _("Create advertiser")),
+            HTML(
+                "<p class='form-text small text-muted'>"
+                + str(
+                    _(
+                        "Creates an advertiser, a campaign, user account, and initial flight. "
+                        "The user will receive an invitation email."
+                    )
+                )
+                + "</p>"
+            ),
+        )
+
+    def clean_advertiser_name(self):
+        advertiser_name = self.cleaned_data["advertiser_name"].strip()
+
+        if Advertiser.objects.filter(name=advertiser_name).exists():
+            raise forms.ValidationError(_("Advertiser already exists"))
+
+        return advertiser_name
+
+    def create_user(self):
+        """Create the user account and send an invite email."""
+        user_name = self.cleaned_data["user_name"].strip()
+        user_email = self.cleaned_data["user_email"]
+        user = User.objects.create_user(name=user_name, email=user_email, password="")
+        if hasattr(user, "invite_user"):
+            user.invite_user()
+        return user
+
+    def create_advertiser(self):
+        """Create the advertiser, campaign, and initial flight."""
+        advertiser_name = self.cleaned_data["advertiser_name"].strip()
+
+        advertiser = Advertiser.objects.create(
+            name=advertiser_name, slug=slugify(advertiser_name)
+        )
+        campaign = Campaign.objects.create(
+            advertiser=advertiser,
+            name=advertiser_name,
+            slug=slugify(advertiser_name),
+        )
+        for pub_group in PublisherGroup.objects.all():
+            campaign.publisher_groups.add(pub_group)
+
+        flight_name = f"{advertiser_name} Initial Flight"
+        Flight.objects.create(
+            campaign=campaign,
+            name=flight_name,
+            slug=slugify(flight_name),
+            cpm=self.DEFAULT_CPM,
+            sold_impressions=self.DEFAULT_NUM_IMPRESSIONS,
+            targeting_parameters={
+                "include_countries": self.DEFAULT_COUNTRY_TARGETING,
+            },
+        )
+
+        return advertiser
+
+    def create_stripe_customer(self, user, advertiser):
+        """Setup a Stripe customer for this user."""
+        if not settings.STRIPE_SECRET_KEY:
```

Comment:
Is it worth logging in this case? Probably not, but not sure.

Reply:
I don't think it's necessary. This will be logged in a number of places since a few other things fail if the Stripe key isn't set.